saleor/graphql/product/tests/test_product.py

```diff
@@ -1761,6 +1768,50 @@ def test_products_query_with_filter(
     assert products[0]["node"]["name"] == second_product.name
 
 
+def test_products_query_with_price_filter_as_staff(
+    query_products_with_filter,
+    staff_api_client,
+    product_list,
+    permission_manage_products,
+    channel_USD,
+):
+    product = product_list[0]
+    product.variants.first().channel_listings.filter().update(price_amount=None)
+
+    variables = {
+        "filter": {"price": {"gte": 9, "lte": 31}, "channel": channel_USD.slug},
+        "channel": channel_USD.slug,
+    }
+    staff_api_client.user.user_permissions.add(permission_manage_products)
+    response = staff_api_client.post_graphql(query_products_with_filter, variables)
+    content = get_graphql_content(response)
+    products = content["data"]["products"]["edges"]
+
+    assert len(products) == 3
+
+
+def test_products_query_with_price_filter_as_user(
+    query_products_with_filter,
+    user_api_client,
+    product_list,
+    permission_manage_products,
+    channel_USD,
+):
+    product = product_list[0]
+    product.variants.first().channel_listings.filter().update(price_amount=None)
+
+    variables = {
+        "filter": {"price": {"gte": 9, "lte": 31}, "channel": channel_USD.slug},
+        "channel": channel_USD.slug,
+    }
+    user_api_client.user.user_permissions.add(permission_manage_products)
+    response = user_api_client.post_graphql(query_products_with_filter, variables)
+    content = get_graphql_content(response)
+    products = content["data"]["products"]["edges"]
+
+    assert len(products) == 2
+
```

Comment:
IMO we should also check if the product with no `price_amount` isn't in the returned products list.